src/main/java/com/google/api/codegen/ApiaryConfig.java

```diff
@@ -230,4 +244,29 @@ public Field getField(Type type, String fieldName) {
   public ListMultimap<String, String> getAuthScopes() {
     return authScopes;
   }
+
+  /*
+   * @return auth type supported by the method.
+   */
+  public AuthType getAuthType(String method) {
```

Comment:
@vchudnov-g PTAL at this method - it determines the auth type to use given a method.

I've opted for an enum instead of a separate class, I think it's a bit simpler.